src/spring-cloud/azext_spring_cloud/_app_managed_identity_validator.py

```diff
@@ -0,0 +1,136 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+from azure.cli.core.azclierror import InvalidArgumentValueError
+from azure.mgmt.core.tools import is_valid_resource_id
+from knack.log import get_logger
+
+
+logger = get_logger(__name__)
+
+
+OBSOLETE_APP_IDENTITY_REMOVE = "Remove managed identities without \"system-assigned\" or \"user-assigned\" parameters is obsolete, will only remove system-assigned managed identity, and will not be supported in the future."
+WARNING_NO_USER_IDENTITY_RESOURCE_ID = "No resource ID of user-assigned managed identity is given for parameter \"user-assigned\", will remove ALL user-assigned managed identities."
+OBSOLETE_APP_IDENTITY_ASSIGN = "Assign managed identities without \"system-assigned\" or \"user-assigned\" parameters is obsolete, will only enable system-assigned managed identity, and will not be supported in the future."
+ENABLE_LOWER = "enable"
+DISABLE_LOWER = "disable"
+
+def validate_app_identity_remove_or_warning(namespace):
+    if namespace.system_assigned is None and namespace.user_assigned is None:
+        logger.warning(OBSOLETE_APP_IDENTITY_REMOVE)
+    if namespace.user_assigned is not None:
+        if not isinstance(namespace.user_assigned, list):
+            raise InvalidArgumentValueError("Parameter value for \"user-assigned\" should be empty or a list of space-separated managed identity resource ID.")
+        if len(namespace.user_assigned) == 0:
+            logger.warning(WARNING_NO_USER_IDENTITY_RESOURCE_ID)
+        namespace.user_assigned = _normalized_user_identitiy_resource_id_list(namespace.user_assigned)
+        for resource_id in namespace.user_assigned:
+            is_valid = _is_valid_user_assigned_managed_identity_resource_id(resource_id)
+            if not is_valid:
+                raise InvalidArgumentValueError("Invalid user-assigned managed identity resource ID \"{}\".".format(resource_id))
+
+
+def _normalized_user_identitiy_resource_id_list(user_identity_resource_id_list):
+    result = []
+    if not user_identity_resource_id_list:
+        return result
+    for id in user_identity_resource_id_list:
+        result.append(id.strip().lower())
+    return result
+
+
+def _is_valid_user_assigned_managed_identity_resource_id(resource_id):
+    if not is_valid_resource_id(resource_id.lower()):
+        return False
+    if "/providers/Microsoft.ManagedIdentity/userAssignedIdentities/".lower() not in resource_id.lower():
```

Comment:
can we use regex and merge the above step as one?

Reply:
`is_valid_resource_id` is provided by azure cli core to validate a general validity of resource ID. And line 48 add an additional check for user identity resource type. Maybe this is easier than we write up a regex.